src/librustc_resolve/lib.rs

```diff
@@ -3024,10 +3019,10 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
             let mut err = this.session.struct_span_err_with_code(base_span, &base_msg, code);
 
             // Emit help message for fake-self from other languages like `this`(javascript)
-            let fake_self: Vec<Ident> = ["this", "my"].iter().map(
+            let mut fake_self = ["this", "my"].iter().map(
                 |s| Ident::from_str(*s)
-            ).collect();
-            if fake_self.contains(&item_str)
+            );
+            if fake_self.any(|ident| ident == *item_str)
```

Comment:
I think you can just use `["this", "my"].contains(&*item_str.as_str())` here.

Reply:
Almost; `["this", "my"].contains(&&*item_str.as_str())`.